python/tskit/tables.py

```diff
@@ -2648,6 +2668,9 @@ def deduplicate_sites(self):
         duplicate ``position`` (and keeping only the *first* entry for each
         site), and renumbering the ``site`` column of the mutation table
         appropriately.  This requires the site table to be sorted by position.
+
+        This method does not sort the tables afterwards, so mutations may no longer
+        be sorted by time.
```

Comment:
I think we may need to do something about this, but it's a job for a different PR.